drivers/lvutil.py

```diff
@@ -725,3 +728,23 @@ def removeDevMapperEntry(path, strict=True):
         util.SMlog("removeDevMapperEntry: dmsetup remove failed for file %s " \
                    "with error %s, and lsof ret is %s." % (path, str(e), ret))
         return False
+
+def vgIsFullyProvisioned(vgname):
+    """
+    Check the PVs behind the specified VG and return
+    True if any of them is fully provisioned
+    """
+    text = cmd_lvm([CMD_VGDISPLAY, '-v', vgname])
```

Comment:
Is there an easier way to find underlying devices? Can we use self.root?

Reply:
hmm, I see that getting from lvutil to LVHDSR is going to be convoluted too. Let's go with this approach then.